test/synth/synth-stdout.test.ts

```diff
@@ -413,3 +1114,47 @@ function requireSynth() {
   // eslint-disable-next-line
   return require(module);
 }
+
+async function expectSynthMatchSnapshot(workdir: string) {
+  const omitted = '__omitted__';
+
+  const files = await promisify(glob)('**', {
+    cwd: workdir,
+    nodir: true,
+  });
+
+  const map: Record<string, string> = {};
+
+  for (const file of files) {
+    const filePath = path.join(workdir, file);
+
+    if (file === CHART_YAML) {
+      const source = fs.readFileSync(filePath, 'utf-8');
+      const yamlContents = yaml.parse(source);
+
+      expect(yamlContents.name).toMatch(/cdk8s-/);
+      expect(yamlContents.description).toMatch(/Generated chart for cdk8s-/);
+
+      // These values are not stable
+      yamlContents.name = omitted;
+      yamlContents.description = omitted;
+
+      map[file] = yaml.stringify(yamlContents);
+      continue;
+    }
+
+    if (file.includes('crds/')) {
+      // crds/encodedFileName are not stable
```

Comment:
How come? shouldn't this only be derived from the CRD file name?

Reply:
This is very strange. The filename does not change when running synth multiple times locally. This is changed when running build on PR: [`be1fe8d` (#1195)](https://github.com/cdk8s-team/cdk8s-cli/pull/1195/commits/be1fe8d0a636474fad18c00ae21b048b5bd91ac5)

Taking a look.

Reply:
So this was happening because the hash was being calculated for the file with its path. This changes when being run on PR.

I have changed it to just hash and encode for file name if it is not a url.